systems/framework/lcm_driven_loop.h

```diff
@@ -114,26 +114,42 @@ class LcmDrivenLoop {
     active_channel_ = active_channel;
   };
 
+  /// Constructor for single-input LcmDrivenLoop without lcm_parser
+  ///     @param drake_lcm DrakeLcm
+  ///     @param diagram A Drake diagram
+  ///     @param input_channel The name of the input channel
+  /// The use case is that the user only need the time from lcm message.
+  LcmDrivenLoop(drake::lcm::DrakeLcm* drake_lcm,
+                std::unique_ptr<drake::systems::Diagram<double>> diagram,
+                const std::string& input_channel)
+      : LcmDrivenLoop(drake_lcm, std::move(diagram), nullptr,
+                      std::vector<std::string>(1, input_channel), input_channel,
+                      ""){};
+
+  // Start simulating the diagram
+  void Simulate() { Simulate(std::numeric_limits<double>::infinity(), true); }
+
   // Start simulating the diagram
-  void Simulate(double end_time = std::numeric_limits<double>::infinity()) {
+  /// @param is_end_time_or_duration, true if `time_in` is the end time of the
```

Comment:
I made the old `Simulate()` function private and created public functions `SimulateTo` and `SimulateDuration`

Reply:
I feel like this approach is still awkward (it's the same code, just hidden now). Is there a reason you think the approach I suggested is worse?

Reply:
I agree that the approach looks awkward, and since we are not using it now, I think I'll just delete it.

The reason I created `delta T` aside from the original `final T` was because the old version of`run_controller_switch.cc` used `LcmDrivenLoop` and we want to turn off `run_controller_switch.cc` after a duration `delta T`.

As for the API, the design arises from minimizing duplicated code. Since the difference between those two modes is just one line code, I decided to leave most of the old `Simulate()` method untouched, and created two functions that call it.